source/Halibut/Diagnostics/CachingLogFactory.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Concurrent;
+using Halibut.Diagnostics.LogCreators;
+
+namespace Halibut.Diagnostics
+{
+    public class CachingLogFactory : ILogFactory
```

Comment:
This is copied from `LogFactory` which did two things cache and log to a in memory log.